internal/pkg/composable/providers/host/host_test.go

```diff
@@ -167,3 +168,20 @@ func returnHostMapping(log *logger.Logger) infoFetcher {
 		return host, nil
 	}
 }
+
+func TestGetHostInfoReturnsSomeKeys(t *testing.T) {
+	// this is a simple test to ensure the host provicer returns the new keys
+	// needed to add support to Debian 12.
+	osInfo, err := getHostInfo(logp.L())()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	expectedKeys := []string{"os_family", "os_platform", "os_version"}
+
+	for _, key := range expectedKeys {
+		if _, exist := osInfo[key]; !exist {
+			t.Errorf("expecting key '%s' from host provider.", key)
+		}
+	}
```

Comment:
I'm not sure if this is needed, but should we also check that the values are not empty?

Reply:
That's highly dependant on the OS/Linux distro it's running on, for the few I've tested, only Arch had an empty string, but overall it's very inconsistent across different distros (some examples: https://github.com/elastic/integrations/issues/10797#issuecomment-2452212459).